webapp/tests/forms/flows/test_multistep_flow.py

```diff
@@ -114,39 +114,37 @@ def test_if_form_step_after_render_step_then_keep_data_from_older_form_step(self
 
     def test_update_session_data_is_called(self):
         expected_data = {'brother: Luigi'}
-        with app.app_context() and app.test_request_context():
-            with patch('app.forms.flows.multistep_flow.override_session_data') as update_fun, \
-                 patch('app.forms.flows.multistep_flow.MultiStepFlow._handle_specifics_for_step',
-                       MagicMock(return_value=(
-                               RenderInfo(step_title="Any", step_intro="Introduction", form=MockForm, prev_url="Prev",
-                                          next_url="Next", submit_url="Submit", overview_url="Overview"), expected_data))):
-                self.flow.handle(MockRenderStep.name)
+        with patch('app.forms.flows.multistep_flow.override_session_data') as update_fun, \
+                patch('app.forms.flows.multistep_flow.MultiStepFlow._handle_specifics_for_step',
+                    MagicMock(return_value=(
+                            RenderInfo(step_title="Any", step_intro="Introduction", form=MockForm, prev_url="Prev",
+                                        next_url="Next", submit_url="Submit", overview_url="Overview"), expected_data))):
+            self.flow.handle(MockRenderStep.name)
 
-                update_fun.assert_called_once_with(expected_data)
+            update_fun.assert_called_once_with(expected_data)
 
     def test_yes_no_field_content_overriden_if_empty(self):
         steps = [MockYesNoStep, MockFinalStep]
-        with app.test_request_context():
-            flow_with_yes_no_field = MultiStepFlow('No_maybe', endpoint=self.endpoint_correct, steps=steps)
-        resulting_session = self.run_handle(flow_with_yes_no_field, MockYesNoStep.name, 'POST', {'yes_no_field': 'yes'})
-        resulting_session = self.run_handle(flow_with_yes_no_field, MockYesNoStep.name, 'POST', {}, resulting_session)
+        flow_with_yes_no_field = MultiStepFlow('No_maybe', endpoint=self.endpoint_correct, steps=steps)
+        resulting_session = self.run_handle(self.app, flow_with_yes_no_field, MockYesNoStep.name, 'POST', {'yes_no_field': 'yes'})
+        resulting_session = self.run_handle(self.app, flow_with_yes_no_field, MockYesNoStep.name, 'POST', {}, resulting_session)
         self.assertEqual({'yes_no_field': None}, deserialize_session_data(resulting_session['form_data'],
-                                                                        app.config['PERMANENT_SESSION_LIFETIME']))
+                                                                        self.app.config['PERMANENT_SESSION_LIFETIME']))
 
     @staticmethod
-    def run_handle(flow, step_name, method='GET', form_data=None, session=None):
-        with app.app_context() and app.test_request_context(method=method) as req:
+    def run_handle(app, flow, step_name, method='GET', form_data=None, session=None):
+        with app.test_request_context(method=method) as req:
```

Comment:
If we do not use a static method, we can directly use self.app or even self.req. I guess that we have to create a new request_context for each method though, correct?

Reply:
I'm not sure why this is done this way actually, I was just trying to keep the tests running as close to what we had before… I've not used `test_request_context` much, I'm more used to using a test client (like most of the tests here: https://flask.palletsprojects.com/en/2.0.x/testing/) for more of a "functional" testing approach with patching to mock resources, but that's a bit hard to do with our code.